src/contract/ga/index.ts

```diff
@@ -0,0 +1,266 @@
+/*
+ * ISC License (ISC)
+ * Copyright (c) 2022 aeternity developers
+ *
+ *  Permission to use, copy, modify, and/or distribute this software for any
+ *  purpose with or without fee is hereby granted, provided that the above
+ *  copyright notice and this permission notice appear in all copies.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
+ *  REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
+ *  AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
+ *  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
+ *  LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
+ *  OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
+ *  PERFORMANCE OF THIS SOFTWARE.
+ */
+
+/**
+ * Generalized Account module - routines to use generalized account
+ *
+ * @module @aeternity/aepp-sdk/es/contract/ga
+ * @export GeneralizedAccount
+ * @example import { GeneralizedAccount } from '@aeternity/aepp-sdk'
+ */
+// @ts-expect-error TODO: remove me
+import Ae from '../../ae'
+import Contract, { _Contract } from '../../ae/contract'
+// @ts-expect-error TODO: remove me
+import { TX_TYPE } from '../../tx/builder/schema'
+// @ts-expect-error TODO: remove me
+import { buildTx, unpackTx } from '../../tx/builder'
+import { prepareGaParams } from './helpers'
+import { hash } from '../../utils/crypto'
+import { decode, EncodedData } from '../../utils/encoder'
+import { IllegalArgumentError, MissingParamError } from '../../utils/errors'
+
+export interface AuthData {
+  gasLimit: number
+  callData: string
+  source?: string
+  args?: object
+}
+
+interface GAObject {
+  owner: string
+  transaction: string
+  rawTx: string
+  gaContractId: string
+}
+
+interface Account {
+  authFun: string
+  balance: string
+  contractId?: string
+  id?: string
+  kind: string
+  nonce: number
+  payable: true
+}
+
+export abstract class _GeneralizedAccount extends _Contract {
+  /**
+   * @alias module:@aeternity/aepp-sdk/es/contract/ga
+   * Check if account is GA
+   * @param address - Account address
+   * @returns if account is GA
+   */
+  async isGA (address: string): Promise<boolean> {
+    const { contractId } = await this.getAccount(address)
+    return Boolean(contractId)
```

Comment:
can this check be more specific? like
```suggestion
    return contractId != null
```